lib/python/Jetson/GPIO/gpio_event.py

```diff
@@ -245,124 +355,52 @@ def _poll_thread():
-        _mutex.acquire()
-        gpio_obj = _get_gpio_object(gpio)
-        _set_edge(gpio_name, edge)
-        gpio_obj.edge = edge
-        gpio_obj.bouncetime = bouncetime
-        gpio_obj.initial_wait = True
-        _gpio_event_list[gpio] = gpio_obj
-        _mutex.release()
 
-    # create epoll blocking object if not already created
-    if _epoll_fd_blocking is None:
-        _epoll_fd_blocking = epoll()
-        if _epoll_fd_blocking is None:
-            return -2
+    thread.exit()
 
-    # register gpio value fd with epoll
+# This function waits for a edge event in a blocking mode, which the user must 
+# specify the file descriptor of the chip, which channel of the chip, the event handle,
+# time for debouncing in milliseconds, the time limit to wait for the event
+def blocking_wait_for_edge(chip_fd, channel, request, bouncetime, timeout):
     try:
-        _epoll_fd_blocking.register(gpio_obj.value_fd, EPOLLIN | EPOLLET |
-                                    EPOLLPRI)
-    except IOError:
-        print("IOError occured while register epoll blocking for GPIO %s"
-              % gpio)
-        return -2
+        fcntl.ioctl(chip_fd, cdev.GPIO_GET_LINEEVENT_IOCTL, request)
+    except (OSError, IOError) as e:
+        raise cdev.GPIOError(e.errno, "Opening input line event handle: " + e.strerror)
 
-    while not finished:
-        # retry polling if interrupted by signal
+    cdev.get_value(request.fd)
+
+    ret = select.select([request.fd], [], [], timeout)
+    if ret[0] == [request.fd]:
         try:
-            res = _epoll_fd_blocking.poll(timeout, maxevents=1)
-        except InterruptedError:
-            continue
+            data = os.read(request.fd, ctypes.sizeof(cdev.gpioevent_data))
+        except OSError as e:
+            raise cdev.GPIOError(e.errno, "Reading GPIO event: " + e.strerror)
 
-        # First trigger is with current state so ignore
-        if initial_edge:
-            initial_edge = False
-            continue
+        event_data = cdev.gpioevent_data.from_buffer_copy(data)
 
-        # debounce input for specified time
-        else:
-            time = datetime.now()
-            time = time.second * 1E6 + time.microsecond
-            if ((gpio_obj.bouncetime is None) or
-                    (time - gpio_obj.lastcall > gpio_obj.bouncetime * 1000) or
-                    (gpio_obj.lastcall == 0) or (gpio_obj.lastcall > time)):
-                gpio_obj.lastcall = time
-                _mutex.acquire()
-                _gpio_event_list[gpio] = gpio_obj
-                _mutex.release()
-                finished = True
-
-    # check if the event detected was valid
-    if res:
-        fileno = res[0][0]
-        fd = gpio_obj.value_fd
-        if fileno != fd.fileno():
-            _epoll_fd_blocking.unregister(gpio_obj.value_fd)
-            print("File object not found after wait for GPIO %s" % gpio)
-            return -2
+        if event_data.id == cdev.GPIOEVENT_REQUEST_RISING_EDGE:
```

Comment:
can you help me remove this print and return a value similar to the old function